services/QuillLMS/spec/services/classlink_integration/classroom_persister_spec.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ClasslinkIntegration::ClassroomPersister do
+  let(:incoming_classroom) do
+    {
+      sourced_id: 'external-123',
+      name: 'Math 101',
+      visible: true
+    }
+  end
+
+  describe '#run' do
+    let(:classlink_instance) { create(:classlink_instance) }
```

Comment:
Since the setup for persister is the same parameters, you could refactor out a 
`subject { described_class.new(incoming_classroom:, existing_provider_classrooms:, classlink_instance_id: classlink_instance.id)`

Then your two tests become:

`expect { subject }`

This communicates that you're testing the same structure of subject in each test.

Reply:
In this case, the persister does not have the same params for each test. This is missable due to the use of the implicit `key:` syntax